front/src/modules/simulationResult/components/SpeedSpaceChart/SpeedSpaceChart.tsx

```diff
@@ -167,6 +194,13 @@ export default function SpeedSpaceChart({
       } else {
         createChartAndTrain();
       }
+      if (chart && setSharedXScaleDomain) {
+        setSharedXScaleDomain((prevState) => ({
+          ...prevState,
+          current: prevState.initial,
+          source: undefined,
+        }));
+      }
```

Comment:
You can simply do this if you want to reset the charts. As you update sharedXScaleDomain, it will retrigger the useEffect above and the graph will be re-drawn
```suggestion
      } else if (chart && setSharedXScaleDomain) {
        setSharedXScaleDomain((prevState) => ({
          ...prevState,
          current: prevState.initial,
          source: undefined,
        }));
      } else {
        createChartAndTrain();
      }
```